nifi-extension-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/credentials/factory/strategies/AbstractServiceAccountCredentialsStrategy.java

```diff
@@ -37,7 +39,13 @@ public AbstractServiceAccountCredentialsStrategy(String name, PropertyDescriptor
 
     @Override
     public GoogleCredentials getGoogleCredentials(Map<PropertyDescriptor, String> properties, HttpTransportFactory transportFactory) throws IOException {
+        final String delegationStrategy = properties.get(CredentialPropertyDescriptors.DELEGATION_STRATEGY);
+        if (delegationStrategy != null && delegationStrategy.equals(DelegationStrategy.DELEGATED_ACCOUNT.getValue())) {
+            final String delegationUser = properties.get(CredentialPropertyDescriptors.DELEGATION_USER);
+            return GoogleCredentials.fromStream(getServiceAccountJson(properties), transportFactory).createDelegated(delegationUser);
+        } else {
             return GoogleCredentials.fromStream(getServiceAccountJson(properties), transportFactory);
+        }
```

Comment:
Given this is an enum and it'll never be null, can't we use the switch case return style here?

Reply:
So that would be possible but the switch statement would be based on the string value since properties map doesn't return the enum directly.  So I'd still have a default case to handle that would essentially be the null check.  

```
switch (delegationStrategy) {
   case DelegationStrategy.DELEGATED_ACCOUNT.getValue():
   ....
   case DelegationStrategy.SERVICE_ACCOUNT.getValue()
   ...
   default:
      throw IllegalArgumentException("Delegation Strategy not defined");
}
```

Reply:
Yeah I thought that we would convert the string to the enum so that we anticipate the case where a value may be added to the enum to support another strategy and in that case the switch case would error as not all values would be covered. But I don't really mind to be honest, I'm ok with all options.